cmd/flag_loader/flag_loader.go

```diff
@@ -0,0 +1,41 @@
+package flag_loader
```

Comment:
nit: I would avoid passing a boolean to the function, as this makes the code harder to debug and maintain.

I'd for something like:

```go
func getOptionalFlagValue(cmd *cobra.Command, flagName, envVarName string) *string {}
func getRequiredFlagValue(cmd *cobra.Command, flagName, envVarName string) (*string, error) {
  // leverage getOptionalFlagValue
}

func GetRequiredRpcUrlFlagValue(cmd *cobra.Command) (*string, error) {}
func GetOptionalRpcUrlFlagValue(cmd *cobra.Command) *string {}

...
```

Reply:
I considered that but tried to avoid creating multiple methods for each flag instead of letting the developer decide it in a function parameter. During the development, the parameter solution seemed to be the right way, but I totally understand your point.

Now that the solution is fully developed, I can focus on these details, I'll try to come up with a different solution.